File: src/main/java/com/gabrielczar/semantic/entities/City.java
package com.gabrielczar.semantic.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Data
@NoArgsConstructor
@EqualsAndHashCode(exclude = {"id", "interestPoints"})
@Entity(name = "cities")
public class City {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(unique = true)
    private String name;

    @JsonBackReference
    @OneToMany(mappedBy = "city", cascade = CascadeType.MERGE)
    private List<InterestPoint> interestPoints;

    public City(String name) {
        this.name = name;
    }

}

File: src/main/java/com/gabrielczar/semantic/entities/City.kt
package com.gabrielczar.semantic.entities;

import com.fasterxml.jackson.annotation.JsonBackReference
import javax.persistence.*

@Entity(name = "cities")
data class City (
        @GeneratedValue(strategy = GenerationType.IDENTITY) @Id val id : Int? = null,

        @Column(unique = true) var name : String,

        @JsonBackReference @OneToMany(mappedBy = "city", cascade = [(CascadeType.MERGE)])
        var interestPoints : MutableList<InterestPoint>? = null
){
    constructor(name : String) : this(id = null, name = name, interestPoints = null)
}
